datadog_checks_dev/datadog_checks/dev/tooling/commands/release.py

```diff
@@ -756,3 +758,84 @@ def freeze(ctx, no_deps):
 
     if not no_deps:
         ctx.invoke(dep_freeze)
+
+
+@release.command(
+    context_settings=CONTEXT_SETTINGS,
+    short_help="Provide a list of the updated checks on a given agent version, in changelog form"
+)
+@click.option('--since', help="Initial Agent version", default='6.3.0')
+@click.option('--to', help="Final Agent version")
+@click.option('--output', '-o', help="Path to the changelog file, if omitted contents will be printed to stdout")
+@click.option('--force', '-f', is_flag=True, default=False, help="Replace an existing file")
+@click.pass_context
+def agent_changelog(ctx, since, to, output, force):
+    """
+    FIXME
+    """
+    agent_tags = git_tag_list(r'^\d+\.\d+\.\d+$')
+
+    # default value for --to is the latest tag
+    if not to:
+        to = agent_tags[-1]
+
+    # filter out versions according to the interval [since, to]
+    agent_tags = [t for t in agent_tags if t >= since and t <= to]
+
+    # reverse so we have descendent order
+    agent_tags = agent_tags[::-1]
+
+    # store the changes in a mapping {agent_version --> {check_name --> current_version}}
+    changes_per_agent = OrderedDict()
+
+    for i in range(1, len(agent_tags)):
+        contents_from = git_show_file(AGENT_REQ_FILE, agent_tags[i-1])
+        catalog_from = parse_agent_req_file(contents_from)
+
+        contents_to = git_show_file(AGENT_REQ_FILE, agent_tags[i])
+        catalog_to = parse_agent_req_file(contents_to)
+
+        changes = OrderedDict()
+        changes_per_agent[agent_tags[i]] = changes
+
+        for name, ver in catalog_to.iteritems():
+            old_ver = catalog_from.get(name, "")
+            if old_ver != ver:
+                # determine whether major version changed
+                breaking = old_ver.split('.')[0] < ver.split('.')[0]
+                changes[name] = (ver, breaking)
+
+    # store the changelog in memory
+    changelog = StringIO()
+
+    # prepare the links
+    agent_changelog_url = 'https://github.com/DataDog/datadog-agent/blob/master/CHANGELOG.rst#{}'
+    check_changelog_url = 'https://github.com/DataDog/integrations-core/blob/master/{}/CHANGELOG.md'
+
+    # go through all the agent releases
+    for agent, changes in changes_per_agent.iteritems():
+        url = agent_changelog_url.format(agent.replace('.', ''))  # Github removes dots from the anchor
+        changelog.write('## Datadog Agent version [{}]({})\n\n'.format(agent, url))
+
+        if not changes:
+            changelog.write('* There were no check updates for this version of the Agent.\n\n')
+        else:
+            for name, ver in changes.iteritems():
+                breaking_notice = " **BREAKING CHANGE** " if ver[1] else ""
+                changelog_url = check_changelog_url.format(name)
+                changelog.write('* {} [{}]({}){}\n'.format(name, ver[0], changelog_url, breaking_notice))
```

Comment:
I understand it may be a little bit more complex to implement. However do we want to write the integration name or the integration folder name. For example `MySQL ` instead `mysql`?

Reply:
Folder